src/aics-image-viewer/components/ViewerStateProvider/index.tsx

```diff
@@ -159,23 +161,22 @@ const ViewerStateProvider: React.FC<{ viewerSettings?: Partial<ViewerState> }> =
     setChannelSettings(newChannelSettings);
   }, []);
 
-  // Sync viewer settings prop with state
-  // React docs seem to be fine with syncing state with props directly in the render function:
-  // https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
-  if (props.viewerSettings) {
-    let newSettings = viewerSettings;
+  useEffect(() => {
+    if (props.viewerSettings) {
+      let newSettings = viewerSettings;
 
-    for (const key of Object.keys(props.viewerSettings) as (keyof ViewerState)[]) {
-      if (viewerSettings[key] !== props.viewerSettings[key]) {
-        // Update viewer settings one at a time to allow change handlers to keep state valid
-        newSettings = applyChangeToViewerSettings(viewerSettings, key, props.viewerSettings[key] as any);
+      for (const key of Object.keys(props.viewerSettings) as (keyof ViewerState)[]) {
+        if (viewerSettings[key] !== props.viewerSettings[key]) {
+          // Update viewer settings one at a time to allow change handlers to keep state valid
+          newSettings = applyChangeToViewerSettings(newSettings, key, props.viewerSettings[key] as any);
```

Comment:
why is newSettings getting reassigned here?

Reply:
`applyChangeToViewerSettings` returns the changed settings object I believe. We have to apply each of the changes one at a time, because we don't want the settings to end up in an invalid state. The change handlers do that for us by blocking certain changes if they would conflict with other settings.